plugins/module_utils/ndb/databases.py

```diff
@@ -0,0 +1,356 @@
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+
+from copy import deepcopy
+
+from ..constants import NDB as NDB_CONSTANTS
+from .clusters import get_cluster_uuid
+from .db_servers import get_db_server_uuid
+from .nutanix_database import NutanixDatabase
+from .profiles import Profile, get_profile_uuid
+from .slas import get_sla_uuid
+from .tags import Tag
+
+__metaclass__ = type
+
+
+class Database(NutanixDatabase):
+    def __init__(self, module):
+        resource_type = "/databases"
+        super(Database, self).__init__(module, resource_type=resource_type)
+        self.build_spec_methods = {
+            "name": self._build_spec_name,
+            "desc": self._build_spec_desc,
+            "auto_tune_staging_drive": self._build_spec_auto_tune_staging_drive,
+            "db_params_profile": self._build_spec_db_params_profile,
+            "db_vm": self._build_spec_db_vm,
+            "time_machine": self._build_spec_time_machine,
+            "postgres": self._build_spec_postgres,
+            "tags": self._build_spec_tags,
+        }
+
+    def get_uuid(
+        self,
+        value,
+        key="name",
+        data=None,
+        entity_type=None,
+        raise_error=True,
+        no_response=False,
+    ):
+        query = {"value-type": key, "value": value}
```

Comment:
query has missing params. For detailed instance of database , query should be like "/databases?detailed=true&load-dbserver-cluster=true"